packages/shared/src/common/file-type.ts

```diff
@@ -0,0 +1,134 @@
+const TIFF_MAGIC_NUMBER_1 = 0x49;
+const TIFF_MAGIC_NUMBER_2 = 0x49;
+const TIFF_MAGIC_NUMBER_3 = 0x2a;
+const TIFF_MAGIC_NUMBER_4 = 0x00;
+const TIFF_MAGIC_NUMBER_5 = 0x4d;
+const TIFF_MAGIC_NUMBER_6 = 0x4d;
+const TIFF_MAGIC_NUMBER_7 = 0x00;
+const TIFF_MAGIC_NUMBER_8 = 0x2a;
+const PDF_MAGIC_NUMBER_1 = 0x25;
+const PDF_MAGIC_NUMBER_2 = 0x50;
+const PDF_MAGIC_NUMBER_3 = 0x44;
+const PDF_MAGIC_NUMBER_4 = 0x46;
+const PDF_MAGIC_NUMBER_5 = 0x2d;
+const XML_MAGIC_NUMBER_1 = 0x3c;
+const XML_MAGIC_NUMBER_2 = 0x3f;
+const XML_MAGIC_NUMBER_3 = 0x78;
+const XML_MAGIC_NUMBER_4 = 0x6d;
+const XML_MAGIC_NUMBER_5 = 0x6c;
+const XML_MAGIC_NUMBER_6 = 0x20;
+const PNG_MAGIC_NUMBER_1 = 0x89;
+const PNG_MAGIC_NUMBER_2 = 0x50;
+const PNG_MAGIC_NUMBER_3 = 0x4e;
+const PNG_MAGIC_NUMBER_4 = 0x47;
+const JPEG_MAGIC_NUMBER_1 = 0xff;
+const JPEG_MAGIC_NUMBER_2 = 0xd8;
+const BMP_MAGIC_NUMBER_1 = 0x42;
+const BMP_MAGIC_NUMBER_2 = 0x4d;
+
+export function isLikelyTextFile(fileBuffer: Buffer): boolean {
+  let readableChars = 0;
+  let nonReadableChars = 0;
+
+  for (let i = 0; i < fileBuffer.length; i++) {
+    const char = fileBuffer[i];
+    if (
+      char !== undefined &&
+      ((char >= 32 && char <= 126) || char === 9 || char === 10 || char === 13)
```

Comment:
we could define consts for these - ie is `9` tab? is it a carriage return?

could also put this code in a shared utility function ie `isValidASCIIChar()` or something

Reply:
`isASCIIChar()` 🤩